uni/lib/view/restaurant/widgets/cantine_slot.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class CantineSlot extends StatelessWidget {
+  final String type;
+  final String name;
+
+  const CantineSlot({
+    Key? key,
+    required this.type,
+    required this.name,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.only(
+      top: 10.0, bottom: 10.0, left: 22.0, right: 22.0),
+      child: Container(
+      key: Key('cantine-slot-type-$type'),
+      child: Row(
+
+        children: [
+          Container(
+              margin: const EdgeInsets.fromLTRB(0, 0, 8.0, 0),
+              child: SizedBox(
+                width: 20,
+                child: createCantineSlotType(context),
+              )),Flexible(
+              child: Text(
+                name,
+                style: Theme.of(context).textTheme.bodyMedium,
+                textAlign: TextAlign.left,
+              )
+          )
+        ],
+      )),
+    );
+  }
+
+  Widget createCantineSlotType(context) {
+    if(type.contains("Carne")) {
```

Comment:
I have a few suggestions to improve this method. You're mapping text to text+tooltip, with the rest of the structure repeated, so you can extract those two variables first and then return the same widget. Then, we should not rely on uppercases: lowercasing the input and comparing to a lowercased string literal seems a bit safer to me, even though no perfect solution exists for this problem.